### PR TITLE
Add deb packaging

### PR DESCRIPTION
**Pull request by keshavdv**

### CODE REVIEW

File: dockerfiles/itest/itest_trusty/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM docker-dev.yelpcorp.com/trusty_yelp
+MAINTAINER John Billings <billings@yelp.com>
+
+RUN apt-get update && apt-get -y install python-dev
+
+ADD itest.sh /itest.sh
+
+RUN chmod +x /itest.sh
```
Comment (solarkennedy) [inline dockerfiles/itest/itest_trusty/Dockerfile]: I would just make the thing executable in the repo?

File: dockerfiles/lucid/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM docker-dev.yelpcorp.com/lucid_yelp
```
Comment (solarkennedy) [inline dockerfiles/lucid/Dockerfile]: Did you intend to use yelp-specific images? I guess for lucid you might have to. :(

File: tox.ini [hunk 3]
```diff
@@ -19,3 +19,39 @@ deps =
...
+    FIG_FILE = dockerfiles/itest/fig_lucid.yml
+commands =
+    fig build
+    fig run itest
+    fig stop
+    fig rm --force
+
+[testenv:itest_trusty]
+setenv =
+    FIG_FILE = dockerfiles/itest/fig_trusty.yml
```
Comment (solarkennedy) [inline tox.ini]: Can you use `docker-compose` words in place of fig?

`COMPOSE_FILE`, `docker-compose up` etc?

Might as well get off on the right foot there right?

File: debian/rules [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/make -f
+# -*- makefile -*-
+
+export DH_OPTIONS
+
+LSBDISTRELEASE := $(shell lsb_release -sr)
+
+ifeq ($(LSBDISTRELEASE),14.04)
+  # wheels built on Lucid won't work for Trusty
+  pip_args=--extra-pip-arg='--no-use-wheel' --no-test
```
Comment (solarkennedy) [inline debian/rules]: If this is public I would almost rather just not use wheels at all

File: debian/rules [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+else
+  pip_args=--no-test
+endif
+
+%:
+	dh $@ --with python-virtualenv
```
Comment (solarkennedy) [inline debian/rules]: It looks like you are not using a specific version of python here, would you like to?

File: debian/control [hunk 6]
```diff
@@ -0,0 +1,8 @@
+Source: hacheck
+Maintainer: John Billings <billings@yelp.com>
+Build-Depends:
+    dh-virtualenv,
+
+Package: hacheck
+Architecture: any
+Description: hacheck
```
Comment (solarkennedy) [inline debian/control]: This needs to explicitly depend on python, preferably the one you are building with

File: dockerfiles/lucid/Dockerfile (hunk 2 above)
Comment (keshavdv) [inline dockerfiles/lucid/Dockerfile]: Yup, none of the Docker Hub one's have any of the required packages.

File: debian/rules [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+override_dh_auto_build:
+	true
+
+# do not call `make test` as part of packaging
+override_dh_auto_test:
+	true
+
+override_dh_virtualenv:
+	dh_virtualenv --extra-index-url 'https://pypi.yelpcorp.com/simple' --python=/usr/bin/python2.7 --no-test
```
Comment (solarkennedy) [inline debian/rules]: This works outside of yelp?

File: debian/control [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Source: hacheck
+Maintainer: John Billings <billings@yelp.com>
+Build-Depends:
+    dh-virtualenv, python, debhelper (>= 7)
```
Comment (solarkennedy) [inline debian/control]: do python2.7 if you are using python2.7 in the rules file

**Comment by solarkennedy** [comment]
ship

File: debian/rules (hunk 7 above)
Comment (keshavdv) [inline debian/rules]: Copied from the paasta repo so I believe it does

**Pull request merged**